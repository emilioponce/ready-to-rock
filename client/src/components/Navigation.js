import React from 'react'
import PropTypes from 'prop-types'

import styles from './Navigation.module.css'

const Navigation = ({ handleClick, current, size }) => {
  const renderPrevButton = () => {
    if (current !== 0) {
      return (
        <button id="prev" className={styles.button} onClick={handleClick}>
          Prev
        </button>
      )
    }
  }

  const renderNextButton = () => {
    if (current !== size - 1) {
      return (
        <button id="next" className={styles.button} onClick={handleClick}>
          Next
        </button>
      )
    }
  }

  return (
    <div className={styles.navigation}>
      {renderPrevButton()}
      <span className={styles.page}>
        {current + 1} from {size}
      </span>
      {renderNextButton()}
    </div>
  )
}

Navigation.propTypes = {
  current: PropTypes.number.isRequired,
  size: PropTypes.number.isRequired,
  handleClick: PropTypes.func.isRequired
}

export default Navigation
