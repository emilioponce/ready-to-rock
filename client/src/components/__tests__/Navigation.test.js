import React from 'react'
import ReactDOM from 'react-dom'
import Navigation from '../Navigation'
import { shallow } from 'enzyme'

const fn = jest.fn()
const component = <Navigation current={1} size={24} handleClick={fn} />
const wrapper = shallow(component)

afterEach(() => {
  jest.clearAllMocks()
})

describe('Component Navigation', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(component, div)
    ReactDOM.unmountComponentAtNode(div)
  })

  it('snapshot renders correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('prev button is clicked', () => {
    const event = { target: { id: 'prev' } }
    wrapper
      .find('button')
      .first()
      .simulate('click', event)
    expect(fn).toHaveBeenCalled()
    expect(fn).toHaveBeenCalledTimes(1)
    expect(fn).toHaveBeenCalledWith(event)
  })

  it('next button is clicked', () => {
    const event = { target: { id: 'next' } }
    wrapper
      .find('button')
      .at(1)
      .simulate('click', event)
    expect(fn).toHaveBeenCalled()
    expect(fn).toHaveBeenCalledTimes(1)
    expect(fn).toHaveBeenCalledWith(event)
  })

  it('prev button do not exist when it is the first band', () => {
    const component = <Navigation current={0} size={24} handleClick={fn} />
    const wrapper = shallow(component)
    expect(wrapper.exists('#prev')).toBeFalsy()
  })

  it('next button do not exist when it is the last band', () => {
    const component = <Navigation current={23} size={24} handleClick={fn} />
    const wrapper = shallow(component)
    expect(wrapper.exists('#next')).toBeFalsy()
  })
})
