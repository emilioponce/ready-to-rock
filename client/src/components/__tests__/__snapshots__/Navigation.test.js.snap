// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Navigation snapshot renders correctly 1`] = `
<div
  className="navigation"
>
  <button
    className="button"
    id="prev"
    onClick={[MockFunction]}
  >
    Prev
  </button>
  <span
    className="page"
  >
    2
     from 
    24
  </span>
  <button
    className="button"
    id="next"
    onClick={[MockFunction]}
  >
    Next
  </button>
</div>
`;
